src/Assignments/ProblemDetails.tsx

```diff
@@ -146,32 +162,38 @@ export const ProblemDetails: React.FC<ProblemDetailsProps> = ({
                 </>
             )}
             <div className="d-flex">
-                {(() => {
-                    if (_.isNil(grade) || _.isNil(problem)) {
-                        return null;
-                    }
+                <MomentReacter
+                    significantMoments={[endDate, deadDate, solutionsMoment]}
+                    stopMoment={solutionsMoment} // Once solutions are available this timer means nothing
+                    logTag='gradedMessage'
+                >
+                    {(currentMoment) => {
+                        if (_.isNil(grade) || _.isNil(problem)) {
+                            return (<></>);
+                        }
 
-                    let message = null;
-                    if (grade.overallBestScore >= 1) {
-                        message = 'You have completed this problem, your attempts will not be recorded.';
-                    } else if (problem.maxAttempts > 0 && grade.numAttempts >= problem.maxAttempts) {
-                        message = 'You have exceeded the attempt limit. Your attempts on this problem will not be graded but will count toward completion.';
-                    } else if (currentMoment.isBefore(deadDate) && currentMoment.isAfter(endDate)) {
-                        message = 'The topic is past due but partial credit is available. Your attempts will be graded with a penalty.';
-                    } else if (currentMoment.isAfter(solutionsMoment)) {
-                        // TODO get from backend
-                        message = 'Solutions are available, your attempts will not be recorded.';
-                    } else if (currentMoment.isBefore(solutionsMoment) && currentMoment.isAfter(deadDate)) {
-                        message = 'The topic is past due. Your attempts on this problem will not be graded but will count toward completion.';
-                    } else if (grade.overallBestScore < 1 && currentMoment.isBefore(endDate) && (grade.numAttempts < problem.maxAttempts || problem.maxAttempts <= INFINITE_MAX_ATTEMPT_VALUE)) {
-                        // All of these situations should already be handled, just making it more defensive
-                        message = 'Your attempts on this problem will be graded.';
-                    } else {
-                        // TODO remote error logging
-                        message = 'An unknown error has occured and it is unclear if your attempt will be graded.';
-                    }
-                    return message;
-                })()}
+                        let message = null;
+                        if (grade.overallBestScore >= 1) {
+                            message = 'You have completed this problem, your attempts will not be recorded.';
```

Comment:
```suggestion
                            message = 'You have completed this problem. Your attempts will no longer be recorded.';
```

Reply:
`no longer` vs `not` both seem correct, I'll change it though